desktop/core/src/desktop/lib/raz/raz_client.py

```diff
@@ -144,17 +142,64 @@ def check_access(self, method, url, params=None, headers=None, data=None):
   def _handle_raz_req(self, raz_url, request_headers, request_data):
     if self.auth_type == 'kerberos':
       auth_handler = requests_kerberos.HTTPKerberosAuth(mutual_authentication=requests_kerberos.OPTIONAL)
-      raz_req = requests.post(raz_url, headers=request_headers, json=request_data, auth=auth_handler, verify=False)
+      raz_response = self._handle_raz_ha(raz_url, headers=request_headers, data=request_data, auth_handler=auth_handler)
+
     elif self.auth_type == 'jwt':
       jwt_token = fetch_jwt()
-
       if jwt_token is None:
         raise PopupException('Knox JWT is not available to send to RAZ.')
 
       request_headers['Authorization'] = 'Bearer %s' % (jwt_token)
-      raz_req = requests.post(raz_url, headers=request_headers, json=request_data, verify=False)
+      raz_response = self._handle_raz_ha(raz_url, headers=request_headers, data=request_data)
+    
+    if not raz_response:
+      raise PopupException('Failed to receive response from RAZ server.')
+
+    return raz_response
+
+
+  def _handle_raz_ha(self, raz_urls, headers, data, auth_handler=None, verify=False):
+    raz_urls_list = raz_urls.split(',')
+    params = {
+      'headers': headers, 
+      'json': data,
+      'verify': verify
+    }
 
-    return raz_req
+    if auth_handler:
+      params['auth'] = auth_handler
+
+    raz_response = None
+    for r_url in raz_urls_list:
+      r_url = "%s/api/authz/%s/access?doAs=%s" % (r_url.rstrip('/'), self.service, self.username)
```

Comment:
can the raz_url be "http://razurl/"? in that case will this fail?

Reply:
Do you mean the RAZ URL is having an ending slash? If true, then yes it will work for that also.